api/containers.py

```diff
@@ -134,6 +134,8 @@ def _get(self, query, projection, admin_only=False, uid=None):
                     query['permissions'] = {'$elemMatch': {'_id': uid or self.uid, 'site': self.source_site, 'access': 'admin'}}
                 else:
                     query['permissions'] = {'$elemMatch': {'_id': uid or self.uid, 'site': self.source_site}}
+            if self.request.GET.get('public', '').lower() in ('1', 'true'):
+                query['$or'] = [{'public': True}, {'permissions': query.pop('permissions')}]
```

Comment:
I don't understand the {'permissions': query.pop('permissions')} part of this line.